javascript/extractor/src/com/semmle/js/extractor/FileExtractor.java

```diff
@@ -90,6 +155,29 @@ protected boolean contains(File f, String lcExt, ExtractorConfig config) {
       public String toString() {
         return "javascript";
       }
+
+      /** Number of bytes to read from the beginning of a ".js" file to detect if it is a binary file. */
+      private static final int fileHeaderSize = 128;
+
+      /** Computes if `f` is a binary file based on whether the initial `fileHeaderSize` bytes are printable UTF-8 chars. */
+      private boolean isBinaryFile(File f, String lcExt, ExtractorConfig config) {
+        try (FileInputStream fis = new FileInputStream(f)) {
+          byte[] bytes = new byte[fileHeaderSize];
+          int length = fis.read(bytes);
+
+          if (length == -1) return false;
+
+          // Avoid invalid or unprintable UTF-8 files.
+          if (config.getDefaultEncoding().equals(UTF8_CHARSET.name()) && hasUnprintableUtf8(bytes, length)) {
```

Comment:
So if `config.getDefaultEncoding().equals(UTF8_CHARSET.name()) == false`, then this method always returns `false` without reading any bytes. Can we move that check outside the `try`-block?